src/nxt_conf_validation.c

```diff
@@ -2913,13 +2912,15 @@ nxt_conf_vldt_object_iterator(nxt_conf_validation_t *vldt,
 
     for ( ;; ) {
         member = nxt_conf_next_object_member(value, &name, &index);
-
         if (member == NULL) {
             return NXT_OK;
         }
 
-        ret = validator(vldt, &name, member);
+        if (nxt_slow_path(nxt_str_dup(vldt->pool, &str, &name) == NULL)) {
+            return NXT_ERROR;
+        }
 
+        ret = validator(vldt, &str, member);
```

Comment:
I suppose this will be duping all object members...

Reply:
Hunk:

```diff
@@ -2913,13 +2912,15 @@ nxt_conf_vldt_object_iterator(nxt_conf_validation_t *vldt,
```

Reply:
I encourage you to test memory usage before and after this patch, to make sure it's as we expect.

Reply:
My feeling (based on the past 18 months) is this is going to be a controversial change (tbh I'm not even completely fond of it).

For the best chances of this being accepted I would not do this change,, but rather just do the strdups in the required places to fix this issue... yes, may be a _little_ extra code, but shouldn't be too bad with your new function...

Reply:
> My feeling (based on the past 18 months) is this is going to be a controversial change (tbh I'm not even completely fond of it).
> 
> For the best chances of this being accepted I would not do this change,, but rather just do the strdups in the required places to fix this issue... yes, may be a _little_ extra code, but shouldn't be too bad with your new function...

I see. My mistake that duplicating will be applied. Thanks for correcting me.
This new way makes sense.